### ISSUE TITLE
Set configuration in etcd for services?

### ISSUE THREAD

**Issue by bmorton** [bmorton/deployster#16]
Is it reasonable for deployster to implement functionality to ease in setting/managing environment variables that are exposed to services by convention?

**Comment by ivanvanderbyl** [comment]
Thought: We could pull the keys from `/services/{name}/{ENV_VARIABLE}` and apply them to the environment supplied in the unit file for the docker container.

**Comment by dcondomitti** [comment]
https://github.com/dcondomitti/bridge

**Comment by ivanvanderbyl** [comment]
@dcondomitti I like that a lot.

**Comment by bobbytables** [comment]
That's actually pretty cool...

**Comment by bmorton** [comment]
Yeah, really cool.  I started using it in my template for this stuff.  Here's what my Dockerfile looks like currently:

```
FROM golang
MAINTAINER Brian Morton "brian@mmm.hm"

ENV APP_NAME myapp
ENV REPO github.com/bmorton/${APP_NAME}

ENV BRIDGE_VERSION v0.0.2-dev
ADD https://github.com/bmorton/bridge/releases/download/${BRIDGE_VERSION}/bridge-linux-amd64 /bin/bridge
RUN chmod +x /bin/bridge

ADD . /go/src/$REPO
RUN cd /go/src/$REPO && go get -v -d
RUN go install $REPO

EXPOSE 3000
ENTRYPOINT exec /bin/bridge -debug -path="/services/${APP_NAME}" -etcd_host=${ETCDCTL_PEERS} /go/bin/scaler
CMD /go/bin/${APP_NAME}
```

**Comment by bmorton** [comment]
Given the bridge stuff, does it make sense that deployster would handle this or should it just be responsible for making sure to inject `ETCDCTL_PEERS` or whatever we want to call it?

@dcondomitti what do you think about changing `-etcd_host` to either take an array of etcd peers or be definable via some env variable?  I like `ETCDCTL_PEERS` because that's what `etcdctl` uses, but it doesn't make much sense if you're not using `etcdctl`.  Something generic would make sense from deployster's perspective though.

**Comment by bobbytables** [comment]
`ETCDCTL_PEERS` makes sense to me.

**Comment by ivanvanderbyl** [comment]
@bmorton @dcondomitti switch the CLI to use https://github.com/codegangsta/cli — it supports EnvFlags:

```go
var (
	etcdHostFlag = cli.StringFlag{
		Name:   "etcd-host",
		Value:  "127.0.0.1",
		EnvVar: "ETCDCTL_PEERS",
	}
)

```

**Comment by bobbytables** [comment]
There's also envflags https://github.com/ianschenck/envflag/tree/master

**Comment by bmorton** [comment]
:+1: I use codegangsta/cli for [`deployctl`](https://github.com/bmorton/deployctl), if you want to use that as an example, @dcondomitti.
